### PR TITLE
feat: re-introduce the OCI environment

### PR DESCRIPTION
**Pull request by cjdcordeiro**
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

This PR brings back the `env` field, now renamed to `environment` and converted into a Dict instead of a List (to be consistent with the `services` syntax.

### CODE REVIEW

File: docs/reference/rockcraft.yaml.rst [hunk 1]
```diff
@@ -111,6 +111,16 @@ The default OCI user. It must be a supported shared user. Currently, the only
 supported shared user is "_daemon_" (with UID/GID 584792). It defaults to
 "root" (with UID 0).
 
+``environment``
+------------
```
Comment (SamirPS) [inline docs/reference/rockcraft.yaml.rst:115]: For the error on the docs workflow just add more ``-`` here

**Review by jnsgruk — COMMENTED** [review]
Nice, thanks @cjdcordeiro :)

**Review by jnsgruk — APPROVED** [review]
Nice, thanks @cjdcordeiro :)

**Review by SamirPS — APPROVED** [review]

File: rockcraft/project.py [hunk 2]
```diff
@@ -228,6 +228,7 @@ class Project(YamlModel):
     platforms: Dict[str, Any]
     base: Literal["bare", "ubuntu:18.04", "ubuntu:20.04", "ubuntu:22.04"]
     build_base: Optional[Literal["ubuntu:18.04", "ubuntu:20.04", "ubuntu:22.04"]]
+    environment: Optional[Dict[str, Optional[str]]]
```
Comment (sergiusens) [inline rockcraft/project.py:231]: why is the value `Optional[str]` and not `str`? I don't see any examples of that either

**Review by sergiusens — CHANGES_REQUESTED** [review]
Looks good, I just have one nit that could be solved with an answer and/or documentation examples or by making the accepted values more strict

File: rockcraft/project.py (hunk 2 above)
Comment (cjdcordeiro) [reply rockcraft/project.py:231]: basically copied from https://github.com/snapcore/snapcraft/blob/main/snapcraft/projects.py#L447. Might be redundant though.

Comment (cjdcordeiro) [reply rockcraft/project.py:231]: @sergiusens given the snapcraft reference, which solution do you prefer? (I'm ok with removing the `Optional`...in fact, that's what I had initially in https://github.com/canonical/rockcraft/commit/c45a50e58c4546e0bcdadb32b58d3d1fe1eae791#diff-0ea7d1a44d76c7792a852505490ca9e3fb738ec2a7814b051ebc828efb8ceceeR229)

Comment (tigarmo) [reply rockcraft/project.py:231]: What's the meaning of the missing string? E.g. what does

```
environment:
  X:
```

mean for the value of `X` in the final container?

Comment (tigarmo) [reply rockcraft/project.py:231]: I tried it out here and apparently the variable get the literal string `"None"` as value

**Comment by sergiusens** [comment]
I believe this is optional on Snapcraft by mistake.Message ID:
***@***.***>

>

File: rockcraft/project.py (hunk 2 above)
Comment (cjdcordeiro) [reply rockcraft/project.py]: alright, I've removed the `Optional`

**Review by tigarmo — APPROVED** [review]

**Review by sergiusens — APPROVED** [review]

**Comment by sergiusens** [comment]
Nice quality work!

**Pull request merged**